Fix part of #13956: Replaced python_utils.ZIP() with zip()

anurag629 opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #13956.
2. This PR does the following: Replaced python_utils.ZIP() with zip()

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## Proof that changes are correct



## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#instructions-for-making-a-code-change).
- Oppiabot will notify you when you don't add a PR_CHANGELOG label. If you are unable to do so, please @-mention a code owner (who will be in the Reviewers list), or ask on [Gitter](https://gitter.im/oppia/oppia-chat).
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.
- Oppiabot can assign anyone for review/help if you leave a comment like the following: "{{Question/comment}} @{{reviewer_username}} PTAL"
- Some of the e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-your-build-fails).

Commit:
all python_utils.ZIP changed with ZIP

anurag629 committed:
Merge branch 'oppia:develop' into anuragoppia1

vojtechjelinek commented:
@anurag629 Please add a proof of changes, ideally a video of the dev server being started.

vojtechjelinek commented on core/controllers/recent_commits.py:
```diff
@@ -53,7 +53,7 @@ def get(self):
         unique_usernames = user_services.get_usernames(unique_user_ids)
         user_id_to_username = dict(
-            python_utils.ZIP(unique_user_ids, unique_usernames))
+            zip(unique_user_ids, unique_usernames))
```
This can be moved to previous line.

vojtechjelinek commented on core/domain/classifier_services.py:
```diff
@@ -572,7 +572,7 @@ def migrate_state_training_jobs(state_training_jobs_mapping):
 
-        for algorithm_id, job_id in python_utils.ZIP(
+        for algorithm_id, job_id in zip(
                 algorithm_ids_to_add, job_ids):
```
This can be moved to the previous line.

vojtechjelinek commented on core/domain/collection_services.py:
```diff
@@ -1071,7 +1071,7 @@ def compute_collection_contributors_summary(collection_id):
     users_settings = user_services.get_users_settings(contributor_ids)
-    for contributor_id, user_settings in python_utils.ZIP(
+    for contributor_id, user_settings in zip(
             contributor_ids, users_settings):
```
This can be moved to the previous line.

vojtechjelinek commented on core/domain/email_manager_test.py:
```diff
@@ -5499,7 +5499,7 @@ def test_can_users_receive_thread_email(self):
         user_ids = []
-        for user_id, username, user_email in python_utils.ZIP(
+        for user_id, username, user_email in zip(
                 gae_ids, usernames, emails):
```
This can be moved to the previous line.

vojtechjelinek commented on core/domain/exp_services.py:
```diff
@@ -1283,7 +1283,7 @@ def compute_exploration_contributors_summary(exploration_id):
     users_settings = user_services.get_users_settings(contributor_ids)
-    for contributor_id, user_settings in python_utils.ZIP(
+    for contributor_id, user_settings in zip(
             contributor_ids, users_settings):
```
This can be moved to the previous line.

vojtechjelinek commented on core/domain/platform_parameter_domain.py:
```diff
@@ -390,7 +390,7 @@ def _is_first_version_smaller(self, version_a, version_b):
 
-        for sub_version_a, sub_version_b in python_utils.ZIP(
+        for sub_version_a, sub_version_b in zip(
                 version_a, version_b):
```
This can be moved to the previous line.

vojtechjelinek commented on core/domain/skill_fetchers.py:
```diff
@@ -40,7 +40,7 @@ def get_multi_skills(skill_ids, strict=True):
     local_skill_models = skill_models.SkillModel.get_multi(skill_ids)
-    for skill_id, skill_model in python_utils.ZIP(
+    for skill_id, skill_model in zip(
             skill_ids, local_skill_models):
```
This can be moved to the previous line.

vojtechjelinek commented on core/domain/skill_services.py:
```diff
@@ -1039,7 +1039,7 @@ def get_multi_user_skill_mastery(user_id, skill_ids):
 
-    for skill_id, skill_mastery_model in python_utils.ZIP(
+    for skill_id, skill_mastery_model in zip(
             skill_ids, skill_mastery_models):
```
This can be moved to the previous line.

vojtechjelinek commented on core/domain/summary_services.py:
```diff
@@ -261,7 +261,7 @@ def get_exploration_metadata_dicts(exploration_ids, user):
     for (exploration_summary, exploration_rights) in (
-            python_utils.ZIP(
+            zip(
                 exploration_summaries, exploration_rights_objects)):
```
This can be moved to the previous line.

vojtechjelinek commented on core/domain/summary_services.py:
```diff
@@ -323,7 +323,7 @@ def get_displayable_exp_summary_dicts_matching_ids(exploration_ids, user=None):
     for (exploration_summary, exploration_rights) in (
-            python_utils.ZIP(
+            zip(
                 exploration_summaries, exploration_rights_objects)):
```
This can be moved to the previous line.

vojtechjelinek commented on core/domain/user_services.py:
```diff
@@ -161,7 +161,7 @@ def get_users_settings(user_ids, strict=False, include_marked_deleted=False):
     if strict:
-        for user_id, user_settings_model in python_utils.ZIP(
+        for user_id, user_settings_model in zip(
                 user_ids, user_settings_models):
```
This can be moved to the previous line (maybe).

vojtechjelinek commented on core/domain/user_services_test.py:
```diff
@@ -623,7 +623,7 @@ def test_get_user_ids_by_role(self):
         user_ids = []
-        for uid, email, name in python_utils.ZIP(
+        for uid, email, name in zip(
                 auth_ids, user_emails, usernames):
```
This can be moved to the previous line (maybe).

vojtechjelinek commented on core/domain/user_services_test.py:
```diff
@@ -595,7 +595,7 @@ def test_get_usernames_by_role(self):
         user_ids = []
-        for auth_id, email, name in python_utils.ZIP(
+        for auth_id, email, name in zip(
                 auth_ids, user_emails, usernames):
```
This can be moved to the previous line (maybe).

vojtechjelinek commented on core/platform/auth/firebase_auth_services.py:
```diff
@@ -409,13 +409,13 @@ def associate_multi_auth_ids_with_user_ids(
             '{auth_id=%r: user_id=%r}' % (auth_id, user_id)
-            for auth_id, user_id in python_utils.ZIP(
+            for auth_id, user_id in zip(
                 auth_ids, user_id_collisions)
```
This can be moved to the previous line (maybe).

vojtechjelinek commented on core/platform/auth/firebase_auth_services.py:
```diff
@@ -424,7 +424,7 @@ def associate_multi_auth_ids_with_user_ids(
             '{user_id=%r: auth_id=%r}' % (user_id, auth_id)
-            for user_id, auth_id in python_utils.ZIP(
+            for user_id, auth_id in zip(
                 user_ids, auth_id_collisions)
```
This can be moved to the previous line (maybe).

vojtechjelinek commented on core/storage/base_model/gae_models_test.py:
```diff
@@ -220,7 +220,7 @@ def test_put_multi(self) -> None:
         base_models.BaseModel.put_multi(models_2_without_none)
-        for model_id, last_updated in python_utils.ZIP(
+        for model_id, last_updated in zip(
                 model_ids, last_updated_values):
```
This can be moved to the previous line (maybe).

vojtechjelinek commented on core/storage/base_model/gae_models_test.py:
```diff
@@ -233,7 +233,7 @@ def test_put_multi(self) -> None:
         base_models.BaseModel.put_multi(models_3_without_none)
-        for model_id, last_updated in python_utils.ZIP(
+        for model_id, last_updated in zip(
                 model_ids, last_updated_values):
```
This can be moved to the previous line (maybe).

vojtechjelinek commented on core/storage/exploration/gae_models.py:
```diff
@@ -337,7 +337,7 @@ def delete_multi( # type: ignore[override]
 
-            versioned_and_exp_rights_models = python_utils.ZIP(
+            versioned_and_exp_rights_models = zip(
                 versioned_models, exp_rights_models)
```
This can be moved to the previous line (maybe).

vojtechjelinek commented on core/storage/collection/gae_models.py:
```diff
@@ -340,7 +340,7 @@ def delete_multi( # type: ignore[override]
             versioned_models = cls.get_multi(entity_ids, include_deleted=True)
-            for model, rights_model in python_utils.ZIP(
+            for model, rights_model in zip(
                     versioned_models, collection_rights_models):
```
This can be moved to the previous line (maybe).

anurag629 commented:
> @anurag629 Please add a proof of changes, ideally a video of the dev server being started.

Please provide reference how i start dev server

vojtechjelinek commented on scripts/linters/linter_utils.py:
```diff
@@ -91,7 +91,7 @@ def get_from_cache(key, factory):
     defaults = defaults if defaults is not None else ()
     default_func_kwargs = dict(
-        python_utils.ZIP(arg_names[-len(defaults):], defaults))
+        zip(arg_names[-len(defaults):], defaults))
```
This can be moved to the previous line (maybe).

vojtechjelinek reviewed (changes requested):
Thanks! Left a few comments.

anurag629 committed:
Merge branch 'oppia:develop' into anuragoppia1

anurag629 committed:
changed python_utils.ZIP to zip

anurag629 committed:
check failed again corrected

anurag629 committed:
moved to previous line

anurag629 committed:
moved all to previous line

anurag629 commented:
@vojtechjelinek please review

Why lint and e2e test are failing

vojtechjelinek commented:
@anurag629 Please do not resolve comments yourself, just reply to them with "Done" and let the reviewer resolve them. 

e2e tests are probably just flaky

lint checks fail for a reason, please read their output and try to fix them, let me know if you don't know how to fix them

vojtechjelinek reviewed (approved):
LGTM!

vojtechjelinek commented:
@DubeySandeep Can you take a full pass, this is a repetitive change so I think it can be merged after you approve.

vojtechjelinek commented:
@anurag629 Also, the proof of changes is missing, please try to at least start the local dev server and record that it works.

anurag629 commented:
> @anurag629 Also, the proof of changes is missing, please try to at least start the local dev server and record that it works.

Please provide reference how to start local dev server

DubeySandeep commented:
@anurag629 You can find the installation steps here: https://github.com/oppia/oppia/wiki/Installing-Oppia

In general, you can find most of the oppia's development-related information on the wiki page.

anurag629 committed:
Merge branch 'oppia:develop' into anuragoppia1

anurag629 commented:
> @anurag629 You can find the installation steps here: https://github.com/oppia/oppia/wiki/Installing-Oppia
> 
> In general, you can find most of the oppia's development-related information on the wiki page.

@vojtechjelinek @DubeySandeep  i tried many times but not able to run it locally because my system requirements are not satisfied.

vojtechjelinek commented:
> > @anurag629 You can find the installation steps here: [oppia/oppia/wiki/Installing-Oppia](https://github.com/oppia/oppia/wiki/Installing-Oppia)
> > In general, you can find most of the oppia's development-related information on the wiki page.
> 
> @vojtechjelinek @DubeySandeep i tried many times but not able to run it locally because my system requirements are not satisfied.

What is the exact error that you get?

anurag629 committed:
Merge branch 'oppia:develop' into anuragoppia1

anurag629 committed:
Merge branch 'oppia:develop' into anuragoppia1

anurag629 commented:
> > > @anurag629 You can find the installation steps here: [oppia/oppia/wiki/Installing-Oppia](https://github.com/oppia/oppia/wiki/Installing-Oppia)
> > > In general, you can find most of the oppia's development-related information on the wiki page.
> > 
> > 
> > @vojtechjelinek @DubeySandeep i tried many times but not able to run it locally because my system requirements are not satisfied.
> 
> What is the exact error that you get?


![Screenshot (1)_LI](https://user-images.githubusercontent.com/78868769/143545841-4ab7723f-fdd0-4a98-8bb3-1391e8f610eb.jpg)

when i tried to run command  "sudo nano /etc/xrdp/startwm.sh"
It shows Directory '/etc/xrdp' does not exist

anurag629 committed:
Merge branch 'oppia:develop' into anuragoppia1

vojtechjelinek commented:
> > > > @anurag629 You can find the installation steps here: [oppia/oppia/wiki/Installing-Oppia](https://github.com/oppia/oppia/wiki/Installing-Oppia)
> > > > In general, you can find most of the oppia's development-related information on the wiki page.
> > > 
> > > 
> > > @vojtechjelinek @DubeySandeep i tried many times but not able to run it locally because my system requirements are not satisfied.
> > 
> > 
> > What is the exact error that you get?
> 
> ![Screenshot (1)_LI](https://user-images.githubusercontent.com/78868769/143545841-4ab7723f-fdd0-4a98-8bb3-1391e8f610eb.jpg)
> 
> when i tried to run command "sudo nano /etc/xrdp/startwm.sh" It shows Directory '/etc/xrdp' does not exist

Not sure how to fix this one, can you ask on Gitter?

srijanreddy98 committed:
Fixes part of #9749: Migrate few instances of angular-html-bind (#14263)

* Revert "Revert "Fix part of #9749: Create a component that shows rte-output-display and use it instead of angular-html-bind. (#13229)" (#13361)"

This reverts commit 792de4596996527a696d0041e0eff2b8384eda30.

* Fix issues

* Fix spelling mistakes in comment

* Merge with upstream/develop

* Address review comments:
  1. Move oppia-rte-output.spec from the list of ignored files for innerHTML check in eslintrc and move to inline disables.
  2. Change the CODEOWNER listing from oppia-rte-parser.service.ts to oppia-rte-parser.service*.ts to include the spec file.
  3. Add refernce to the note at the top of file for all value.replace ops.
  4. Miscellaneous changes.

Co-authored-by: Vasa Srijan Reddy <redvasa@amazon.com>

anurag629 committed:
Merge branch 'oppia:develop' into anuragoppia1

anurag629 committed:
Merge branch 'oppia:develop' into anuragoppia1

srijanreddy98 reviewed (changes requested):
Hi @anurag629. In order to remove some lines from the final commit message in this PR, I raised this "request changes" comment on this PR so that it can't be merged by anyone else. You don't have to do anything. I will be monitoring the PR and when it has all the approvals and the tests passing I will merge the PR. Please reply to the email I have sent you if you have any questions about this. Thanks -- Srijan

anurag629 committed:
Merge branch 'oppia:develop' into anuragoppia1

anurag629 committed:
Merge branch 'oppia:develop' into anuragoppia1

anurag629 committed:
Merge branch 'oppia:develop' into anuragoppia1

anurag629 committed:
Merge branch 'oppia:develop' into anuragoppia1

anurag629 committed:
Merge branch 'oppia:develop' into anuragoppia1

anurag629 committed:
Merge branch 'oppia:develop' into anuragoppia1

anurag629 commented:
@vojtechjelinek 
link checks are failing. I checked it and  this is due to unused python_utils imported from core in many files.

Should i remove them or not??

vojtechjelinek commented:
> @vojtechjelinek link checks are failing. I checked it and this is due to unused python_utils imported from core in many files.
> 
> Should i remove them or not??

yes, remove them

seanlip reviewed (approved):
Thanks -- LGTM for codeowners.

anurag629 committed:
Merge branch 'oppia:develop' into anuragoppia1

anurag629 committed:
removed all python_utils unused imports

anurag629 committed:
removed all unused python_utils imports

anurag629 committed:
Merge branch 'oppia:develop' into anuragoppia1

anurag629 commented:
@vojtechjelinek  how should i fix these errors

![Screenshot (2)_LI](https://user-images.githubusercontent.com/78868769/145777829-fc92c872-e6db-46d6-8ca8-f786d23f0b1a.jpg)

vojtechjelinek commented:
> @vojtechjelinek how should i fix these errors
> 
> ![Screenshot (2)_LI](https://user-images.githubusercontent.com/78868769/145777829-fc92c872-e6db-46d6-8ca8-f786d23f0b1a.jpg)

The lines are too long, you need to split them into multiple lines so that each can be shorter than 80 chars.

anurag629 committed:
line too long lint checks corrected

SAEb-ai commented:
Unassigning and removing my request for review as I don't think so I have any codeowner files here. Thanks!

anurag629 committed:
Merge branch 'oppia:develop' into anuragoppia1

anurag629 committed:
Merge branch 'oppia:develop' into anuragoppia1

DubeySandeep commented on scripts/pre_push_hook.py:
```diff
@@ -43,7 +43,7 @@
 from scripts import common  # isort:skip  # pylint: disable=wrong-import-position
 from scripts import install_backend_python_libs # isort:skip  # pylint: disable=wrong-import-position
-from core import python_utils  # isort:skip  # pylint: disable=wrong-import-position
+# isort:skip  # pylint: disable=wrong-import-position
```
Remove these unwanted bypass-flags

DubeySandeep replied:
@anurag629 Is this comment resolved? (Please make sure to reply to the comments once it's resolved!)

anurag629 replied:
yes

DubeySandeep reviewed (changes requested):
@anurag629 I've left one comment, PTAL!

Also, are we expecting to remove `ZIP` function from `python_utils.py`?

anurag629 committed:
Merge branch 'oppia:develop' into anuragoppia1

anurag629 committed:
resolved comment by dubey sandeep

anurag629 committed:
fix

anurag629 commented:
@DubeySandeep  review this RR

anurag629 committed:
Merge branch 'oppia:develop' into anuragoppia1

anurag629 committed:
Merge branch 'oppia:develop' into anuragoppia1

anurag629 committed:
Merge branch 'oppia:develop' into anuragoppia1

DubeySandeep commented:
@anurag629 Please make sure that the PR is assigned to reviewers else they won't be able to review it within the expected time! (You can ask OppiaBot to assign a reviewer (check PR pointers above) or you can ask someone on gitter!)

anurag629 committed:
Merge branch 'oppia:develop' into anuragoppia1

anurag629 committed:
Merge branch 'oppia:develop' into anuragoppia1

anurag629 committed:
Merge branch 'oppia:develop' into anuragoppia1

anurag629 committed:
Merge branch 'oppia:develop' into anuragoppia1

anurag629 commented:
> Hi @anurag629. In order to remove some lines from the final commit message in this PR, I raised this "request changes" comment on this PR so that it can't be merged by anyone else. You don't have to do anything. I will be monitoring the PR and when it has all the approvals and the tests passing I will merge the PR. Please reply to the email I have sent you if you have any questions about this. Thanks -- Srijan



> Hi @anurag629. In order to remove some lines from the final commit message in this PR, I raised this "request changes" comment on this PR so that it can't be merged by anyone else. You don't have to do anything. I will be monitoring the PR and when it has all the approvals and the tests passing I will merge the PR. Please reply to the email I have sent you if you have any questions about this. Thanks -- Srijan

Hey @srijanreddy98  i have resolved the changes

anurag629 commented:
@vojtechjelinek how to fix end-to-end e2e test

vojtechjelinek commented:
> @vojtechjelinek how to fix end-to-end e2e test

I have restarted them they were just flaky.

DubeySandeep reviewed (approved):
LGTM for the codeowner files(3)

anurag629 commented:
@vojtechjelinek  all file code owner approved changes. when it will be merged??
